dev/PagerControl/PagerControl.cpp

```diff
@@ -0,0 +1,773 @@
+        }
+        else
+        {
+            winrt::VisualStateManager::GoToState(*this, c_nextPageButtonEnabledVisualState, false);
+        }
+        winrt::VisualStateManager::GoToState(*this, c_lastPageButtonDisabledVisualState, false);
+    }
+    else
+    {
+        winrt::VisualStateManager::GoToState(*this, c_firstPageButtonEnabledVisualState, false);
+        winrt::VisualStateManager::GoToState(*this, c_previousPageButtonEnabledVisualState, false);
+        winrt::VisualStateManager::GoToState(*this, c_nextPageButtonEnabledVisualState, false);
+        winrt::VisualStateManager::GoToState(*this, c_lastPageButtonEnabledVisualState, false);
+    }
+
+    // Handle HiddenOnEdge states
+    if (FirstButtonVisibility() == winrt::PagerControlButtonVisibility::HiddenOnEdge)
+    {
+        if (selectedPageIndex != 0)
+        {
+            winrt::VisualStateManager::GoToState(*this, c_firstPageButtonVisibleVisualState, false);
+        }
+        else
+        {
+            winrt::VisualStateManager::GoToState(*this, c_firstPageButtonHiddenVisualState, false);
+        }
+    }
+    if (PreviousButtonVisibility() == winrt::PagerControlButtonVisibility::HiddenOnEdge)
+    {
+        if (selectedPageIndex != 0)
+        {
+            winrt::VisualStateManager::GoToState(*this, c_previousPageButtonVisibleVisualState, false);
+        }
+        else
+        {
+            winrt::VisualStateManager::GoToState(*this, c_previousPageButtonHiddenVisualState, false);
+        }
+    }
+    if (NextButtonVisibility() == winrt::PagerControlButtonVisibility::HiddenOnEdge)
+    {
+        if (selectedPageIndex != numberOfPages - 1)
+        {
+            winrt::VisualStateManager::GoToState(*this, c_nextPageButtonVisibleVisualState, false);
+        }
+        else
+        {
+            winrt::VisualStateManager::GoToState(*this, c_nextPageButtonHiddenVisualState, false);
+        }
+    }
+    if (LastButtonVisibility() == winrt::PagerControlButtonVisibility::HiddenOnEdge)
+    {
+        if (selectedPageIndex != numberOfPages - 1)
+        {
+            winrt::VisualStateManager::GoToState(*this, c_lastPageButtonVisibleVisualState, false);
+        }
+        else
+        {
+            winrt::VisualStateManager::GoToState(*this, c_lastPageButtonHiddenVisualState, false);
+        }
+    }
+}
+
+/* NumberPanel logic */
+void PagerControl::UpdateNumberPanel(const int numberOfPages)
+{
+    if (numberOfPages < 0)
+    {
+        UpdateNumberOfPanelCollectionInfiniteItems();
+    }
+    else if (numberOfPages < 7)
+    {
+        UpdateNumberPanelCollectionAllItems(numberOfPages);
+    }
+    else
+    {
+        const auto selectedIndex = SelectedPageIndex();
+        // Idea: Choose correct "template" based on SelectedPageIndex (x) and NumberOfPages (n)
+        // 1 2 3 4 5 6 ... n <-- Items
+        if (selectedIndex < 4)
+        {
+            // First two items selected, create following pattern:
```

Comment:
>two [](start = 21, length = 3)

First 3 items

Reply:
Actually four: 0,1,2,3; will update it soon.